src/posthog-featureflags.ts

```diff
@@ -379,14 +399,25 @@ export class PostHogFeatureFlags {
 
         if (flags === false) {
             this.instance.persistence.unregister(PERSISTENCE_OVERRIDE_FEATURE_FLAGS)
-        } else if (isArray(flags)) {
-            const flagsObj: Record<string, string | boolean> = {}
-            for (let i = 0; i < flags.length; i++) {
-                flagsObj[flags[i]] = true
-            }
-            this.instance.persistence.register({ [PERSISTENCE_OVERRIDE_FEATURE_FLAGS]: flagsObj })
+            this.instance.persistence.unregister(PERSISTENCE_OVERRIDE_FEATURE_FLAG_PAYLOADS)
         } else {
-            this.instance.persistence.register({ [PERSISTENCE_OVERRIDE_FEATURE_FLAGS]: flags })
+            if (isArray(flags)) {
+                const flagsObj: Record<string, string | boolean> = {}
+                for (let i = 0; i < flags.length; i++) {
+                    flagsObj[flags[i]] = true
+                }
+                this.instance.persistence.register({ [PERSISTENCE_OVERRIDE_FEATURE_FLAGS]: flagsObj })
+            } else {
+                this.instance.persistence.register({ [PERSISTENCE_OVERRIDE_FEATURE_FLAGS]: flags })
+            }
+
+            if (payloads) {
+                this.instance.persistence.register({ [PERSISTENCE_OVERRIDE_FEATURE_FLAG_PAYLOADS]: payloads })
+            }
+        }
+
+        if (triggerFlagEvent) {
+            this._fireFeatureFlagsCallbacks()
```

Comment:
this should be sufficient to trigger hooks like `useFeatureFlagEnabled`, which is what I was yapping about here: https://posthog.slack.com/archives/C07Q2U4BH4L/p1738088710814869